packages/graphback-core/src/runtime/subscriptionFilter.ts

```diff
@@ -0,0 +1,172 @@
+
+    return parsedFieldValue > parsedFilterValue;
+  },
+  ge: (filterValue: InputType) => (fieldValue: InputType) => {
+    const parsedFieldValue = convertType(fieldValue, typeof filterValue);
+    const parsedFilterValue = convertType(filterValue, typeof parsedFieldValue)
+
+    return parsedFieldValue >= parsedFilterValue;
+  },
+  le: (filterValue: InputType) => (fieldValue: InputType) => {
+    const parsedFieldValue = convertType(fieldValue, typeof filterValue);
+    const parsedFilterValue = convertType(filterValue, typeof parsedFieldValue)
+
+    return parsedFieldValue <= parsedFilterValue;
+  },
+  lt: (filterValue: InputType) => (fieldValue: InputType) => {
+    const parsedFieldValue = convertType(fieldValue, typeof filterValue);
+    const parsedFilterValue = convertType(filterValue, typeof parsedFieldValue)
+
+    return parsedFieldValue < parsedFilterValue;
+  },
+  in: (filterValue: InputType[]) => (fieldValue: InputType) => {
+    return filterValue.map((f: InputType) => f?.toString()).includes(fieldValue?.toString());
+  },
+  between: ([fromVal, toVal]: InputType[]) => (fieldValue: InputType) => {
+    if (isDateObject(fieldValue)) {
+      const fieldValDate = convertType(fieldValue, typeof fieldValue)
+      const fromValDate = convertType(fromVal, typeof fieldValue)
+      const toValDate = convertType(toVal, typeof fieldValue)
+
+      return fieldValDate >= fromValDate && fieldValDate <= toValDate;
+    }
+
+    const parsedFieldValue = Number(fieldValue)
+
+    return parsedFieldValue >= Number(fromVal) && parsedFieldValue <= Number(toVal);
+  },
+  contains: (filterValue: InputType = '') => (fieldValue: InputType = '') => {
+    return fieldValue?.toString().includes(filterValue?.toString());
+  },
+  startsWith: (filterValue: string = '') => (fieldValue: string = '') => {
+    return fieldValue?.toString().startsWith(filterValue?.toString());
+  },
+  endsWith: (filterValue: string = '') => (fieldValue: string = '') => {
+    return fieldValue?.toString().endsWith(filterValue?.toString());
+  }
+}
+
+/**
+ * Dynamically creates a subscription filter predicate using the filter object values
+ * 
+ * @param {QueryFilter} filter - subscription filter input object
+ */
+export function createSubscriptionFilterPredicate<T = any>(filter: QueryFilter): (input: Partial<T>) => boolean {
+  const andFilter = filter.and;
+  const orFilter = filter.or;
+  const notFilter = filter.not;
+
+  const filterFields = Object.keys(filter).filter((key: string) => !['and', 'or', 'not'].includes(key));
+
+  return (payload: Partial<T>): boolean => {
+    let predicateResult = true;
+    for (const fieldName of filterFields) {
+
+      const fieldFilter = filter[fieldName];
+
+      for (const [expr, exprVal] of Object.entries(fieldFilter)) {
+        const predicateFn: PredicateFn = predicateMap[expr](exprVal)
+
+        predicateResult = predicateFn(payload[fieldName])
```

Comment:
```suggestion
        predicateResult = predicateResult && predicateFn(payload[fieldName])
```

Are we missing an `and` operator? 

Also looking at the code, I think it should be possible to short circuit the loop 

```
if (!predicateFn(payload[fieldName])) {
  predicateResult = false;
  break;
}
```

I'll propose treating `or` operator first, since this will enable for further short circuiting e.g instead of a `break;` statement in my suggestion we could `return false` when the `orPredicate` evaluation is false.

Reply:
Yep, `or` is missing now. But will apply your second proposal.